ledger/sandbox-classic/src/main/scala/platform/sandbox/stores/ledger/inmemory/InMemoryLedger.scala

```diff
@@ -353,12 +367,14 @@ private[sandbox] final class InMemoryLedger(
                 transactionId,
                 transaction,
               )
+          val enrichedCommittedTransaction =
```

Comment:
i don’t think always enriching is the right approach. We should do this in the translation if the `verbose` flag is set not always.

Reply:
I did initially try the approach of enriching when returning transactions and contracts form the ledger (conditional on `verbose`), but I couldn't get it to work.

I think the problem is that some of the lookup methods might be dependant on having the in memory store the enriched transaction.

Reply:
Which lookups do you mean and which error did you get when trying that? Doing the enrichment conditionally on the `verbose` flag is what the SQL backend does so I don’t understand why it doesn’t work here.

Reply:
discussion moved to slack

Reply:
changed to the suggested approach